Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -415,6 +454,43 @@ std::string Reader::ReadString() {
   return result;
 }
 
+void EncodeTimestamp(Writer* writer, const Timestamp& timestamp_value) {
+  google_protobuf_Timestamp timestamp_proto =
+      google_protobuf_Timestamp_init_zero;
+  timestamp_proto.seconds = timestamp_value.seconds();
```

Comment:
Instead of initializing to zero and then assigning something else, why not

```c++
google_protobuf_Timestamp timestamp_proto{
  .seconds = timestamp_value.seconds(),
  .nanos = timestamp_value.nanoseconds()
};
```
Actually it looks like designated initializers are a C99 feature not standardized in C++11. Humbug.

Reply:
A slight modification on what you wrote *does* work though:

```
google_protobuf_Timestamp timestamp_proto{
  timestamp_value.seconds(),
  timestamp_value.nanoseconds()
};
```

However, if new fields were added to this struct, c++ wouldn't necessarily catch it. With *our* compiler, we'd be ok, since we define `-Werror=missing-field-initializers`, but I'm unsure about other compilers (eg msvc).

Left as is for now.